src/world/Spell/Customization/SpellCustomizations.cpp

```diff
@@ -87,175 +107,83 @@ void SpellCustomizations::LoadSpellInfoData()
+            {
                 spellInfo.setEffect(dbc_spell_entry->Effect[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectDieSides(dbc_spell_entry->EffectDieSides[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectRealPointsPerLevel(dbc_spell_entry->EffectRealPointsPerLevel[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectBasePoints(dbc_spell_entry->EffectBasePoints[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectMechanic(dbc_spell_entry->EffectMechanic[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectImplicitTargetA(dbc_spell_entry->EffectImplicitTargetA[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectImplicitTargetB(dbc_spell_entry->EffectImplicitTargetB[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectRadiusIndex(dbc_spell_entry->EffectRadiusIndex[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectApplyAuraName(dbc_spell_entry->EffectApplyAuraName[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectAmplitude(dbc_spell_entry->EffectAmplitude[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectMultipleValue(dbc_spell_entry->EffectMultipleValue[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectChainTarget(dbc_spell_entry->EffectChainTarget[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectItemType(dbc_spell_entry->EffectItemType[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectMiscValue(dbc_spell_entry->EffectMiscValue[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
+#if VERSION_STRING >= TBC
                 spellInfo.setEffectMiscValueB(dbc_spell_entry->EffectMiscValueB[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
+#endif
                 spellInfo.setEffectTriggerSpell(dbc_spell_entry->EffectTriggerSpell[j], j);
-
-            for (uint8 j = 0; j < MAX_SPELL_EFFECTS; ++j)
                 spellInfo.setEffectPointsPerComboPoint(dbc_spell_entry->EffectPointsPerComboPoint[j], j);
-
-#if VERSION_STRING > TBC
-            for (uint8 x = 0; x < 3; ++x)
-                for (uint8 j = 0; j < 3; ++j)
-                    spellInfo.setEffectSpellClassMask(dbc_spell_entry->EffectSpellClassMask[x][j], x, j);
+#if VERSION_STRING == WotLK
+                for (auto x = 0; x < 3; ++x)
+                    spellInfo.setEffectSpellClassMask(dbc_spell_entry->EffectSpellClassMask[j][x], j, x);
```

Comment:
Inside of the for loop for effects?

Reply:
Yes, there are three mask fields (all uint32_t) for each spell effect, making it total 9 mask fields. For cata I did the same loop (see https://github.com/AscEmu/AscEmu/pull/578/files/1f15913ed2b2664cd8efc0a488505c713d7fa4be#diff-adb7a1aca5c4d2b402d89c6105df47edR353). In TBC/Classic this field does not exist and we need to figure out how to handle it later... These fields control what spells does the spell modify/affect.